### ISSUE TITLE
Cannot join server

### ISSUE THREAD

**Issue by MjcheI** [CrackyStudio/arkso#18]
Hey there,
I love the tool you created but it doesn't seem to quite work for me.
I have not done anything to the server besides what you can see in the attached screenshots.
Hope you can help!
Thanks

![console](https://user-images.githubusercontent.com/77505032/104759445-3a359880-5760-11eb-8f82-e19fd6288e25.png)
![ark ingame](https://user-images.githubusercontent.com/77505032/104759450-3b66c580-5760-11eb-9bd3-bfbdc2057da2.jpg)

**Comment by CrackyStudio** [comment]
Hello!

The server setup seems good.
Did you allowed 'ShooterGameServer.exe' in your firewall / AV ?

**Comment by MjcheI** [comment]
I just checked and it seems like I did.
When I try to add allow it, it says 'ShooterGame' is already on the list.
Are these two the same?

**Comment by CrackyStudio** [comment]
Not they are not. Allow "ShooterGameServer.exe"

**Comment by MjcheI** [comment]
Like I said, it does not allow that.

**Comment by CrackyStudio** [comment]
This should be possible.
Are you chossing the good file ? 
![image](https://user-images.githubusercontent.com/39296488/104777648-4af40780-577c-11eb-95fb-2e6c4e343b0a.png)

**Comment by MjcheI** [comment]
I was actually in the wrong menu but I went to the right one and it does not work still.
May there be anything weird in the console?

**Comment by CrackyStudio** [comment]
The console said that the server started fine. It's something with right access, so something with firewall I think.

**Comment by MjcheI** [comment]
Would there be any way to reinstall the application without having the options be saved?
I tried reinstalling but it kept the options as I left it.
And the console says 'SteamSocketsOpenSource: gethostname failed ()' and 'gethostbyname failed ()', might that be the issue?

**Comment by CrackyStudio** [comment]
Yes it should. I recently changed my ISP, wiped my PC and bring back anything from 0, and I've exactly the same problem. I really don't know what's the problem and I also try to fix it.

**Comment by CrackyStudio** [comment]
I don't know why, but this setting for query port works (LAN + ONLINE added to favorites) : 
![image](https://user-images.githubusercontent.com/39296488/104920133-a4844e00-5997-11eb-9a40-3e05e3a2924b.png)

Try it, and give feedback please!

**Comment by MjcheI** [comment]
It does actually work now. But it doesn't appear in my favorite, only in LAN. Could my friends still join?
Btw, thank you so much for the programm!

**Comment by CrackyStudio** [comment]
Ok, nice!

If it appears only in LAN, it means that SteamCMD and ARKSO config is now good and the problem is now in port forwarding / firewall configuration.

Allow ShooterGame in both mode (Private/Public) here:
<img width="560" alt="Sans titre 1" src="https://user-images.githubusercontent.com/39296488/104930260-35155b00-59a5-11eb-80a2-6ff1279d957c.png">

Allow ShooterGameServer in Inboud Rules of firewall:
<img width="560" alt="Sans titre 2" src="https://user-images.githubusercontent.com/39296488/104931111-46129c00-59a6-11eb-9390-5b579c0bbdd8.png">

Forward ports like this in your rooter:
Internal Port | External Port | Protocol | IP |
-- | -- | -- | -- |
7777 | 7778 | TCP/UDP | yourserverip |
27015 | 27015 | TCP/UDP | yourserverip |

Then it should works!

If your friend uses Epic Games, add this in ARKSO:
<img width="560" alt="Sans titre 3" src="https://user-images.githubusercontent.com/39296488/104931392-a0136180-59a6-11eb-8b5a-cc240fe0f7f7.png">
(I never tested it, but it should works)

**Comment by MjcheI** [comment]
Last question: Does he also need to open his ports and all that?
Thank you so much for taking your time and helping!!

**Comment by CrackyStudio** [comment]
No, he doesn't need :)

I hope I solved all your problems and you can play!
